### PR TITLE
FED-3584 add to module entity

### PR DESCRIPTION
**Pull request by aid848**
https://desire2learn.atlassian.net/browse/FED-3584

LMS PR: https://github.com/Brightspace/lms/pull/53148

Adds moduleId, orgUnitId, and GenAI url to the module serializer so we can use the GenAI dialog. I added orgUnitId as currently the consumers of this API are using `data-he-context` which [has been discouraged](https://d2l.slack.com/archives/C0PHG3QB0/p1714660400187059)

### CODE REVIEW

File: src/activities/content/ContentModuleEntity.js [hunk 1]
```diff
@@ -62,6 +62,32 @@ export class ContentModuleEntity extends Entity {
...
+
+	/**
+	 * @returns {string} Returns the endpoint for generating a summary for the module
+	 */
+	generateSummaryEndpoint() {
+		if (!this._entity || !this._entity.hasLinkByRel(Rels.Content.generateModuleSummary)) {
+			return;
+		}
+
+		return this._entity.getLinkByRel(Rels.Content.generateModuleSummary).href;
```
Comment (mostlyfabulous) [inline src/activities/content/ContentModuleEntity.js:88]: `generateModuleSummary` or `generateModuleSummaryEndpoint` or `generateSummaryEndpoint`?

File: test/activities/content/ContentModuleEntity.test.js [hunk 2]
```diff
@@ -38,6 +38,18 @@ describe('ContentModuleEntity', () => {
...
+		it('reads org unit id', () => {
+			expect(contentModuleEntity.orgUnitId()).to.equal('6613');
+		});
+
+		it('reads module id', () => {
+			expect(contentModuleEntity.moduleId()).to.equal('12345');
+		});
+
+		it('reads generate module summary url', () => {
+			expect(contentModuleEntity.generateSummaryEndpoint()).to.equal('https://fake-tenant-id.modules.api.proddev.d2l/6613/modules/12345/summary');
```
Comment (mostlyfabulous) [inline test/activities/content/ContentModuleEntity.test.js:51]: Related to first comment

**Review by mdulat — APPROVED** [review]

File: src/activities/content/ContentModuleEntity.js (hunk 1 above)
Comment (aid848) [reply src/activities/content/ContentModuleEntity.js:88]: Hmm yeah, I want it to be somewhat generic but the Rel is currently only in the modules domain.

File: test/activities/content/ContentModuleEntity.test.js (hunk 2 above)
Comment (aid848) [reply test/activities/content/ContentModuleEntity.test.js:51]: Updated

File: src/activities/content/ContentModuleEntity.js (hunk 1 above)
Comment (aid848) [reply src/activities/content/ContentModuleEntity.js]: Updated to get rid of using `module` everywhere. It should be part of the entity name or prefix

**Review by mostlyfabulous — APPROVED** [review]

**Pull request merged**